src/core/p5.Renderer2D.js

```diff
@@ -801,17 +829,30 @@ class Renderer2D extends Renderer {
         return this;
       }
     }
-    if (!this._clipping) ctx.beginPath();
-    ctx.moveTo(x1, y1);
-    ctx.lineTo(x2, y2);
-    ctx.lineTo(x3, y3);
-    ctx.lineTo(x4, y4);
-    ctx.closePath();
-    if (!this._clipping && doFill) {
-      ctx.fill();
-    }
-    if (!this._clipping && doStroke) {
-      ctx.stroke();
+    if (this._clipping) {
+      const tempPath = new Path2D();
+      tempPath.moveTo(x1, y1);
+      tempPath.lineTo(x2, y2);
+      tempPath.lineTo(x3, y3);
+      tempPath.lineTo(x4, y4);
+      tempPath.closePath();
+      const currentTransform = this.drawingContext.getTransform();
+      const ClipBaseTransform = this._clipBaseTransform.inverse();
+      const relativeTransform = ClipBaseTransform.multiply(currentTransform);
+      this.clipPath.addPath(tempPath, relativeTransform);
+    } else{
+      ctx.beginPath();
+      ctx.moveTo(x1, y1);
+      ctx.lineTo(x2, y2);
+      ctx.lineTo(x3, y3);
+      ctx.lineTo(x4, y4);
+      ctx.closePath();
+      if (!this._clipping && doFill) {
```

Comment:
We can take out the `!this._clipping` from this condition since we'll only be in the else block if this is the case